actionpack/lib/action_controller/metal/http_authentication.rb

```diff
@@ -79,11 +79,13 @@ def http_basic_authenticate_with(name:, password:, realm: nil, **options)
         end
 
         def http_basic_authenticate_or_request_with(name:, password:, realm: nil, message: nil)
+          name = name.to_s
+          password = password.to_s
```

Comment:
How about nil -> raise because that might mean you've failed to pull a value out of ENV etc, "" -> acceptable slightly-more-explicit statement that you intend to allow a blank password?

Reply:
Yeah I agrees this isn't ideal, I actually don't want to allow nil when `http_basic_authenticate_or_request_with` is used, only `authenticate_with_http_basic do`.